### PR TITLE
sope-mime: Fixed UTF-8 q-encoding in MIME headers

### PR DESCRIPTION
**Pull request by javierag**
The q-encoding was broke because it encoded the whole
header line; this made MIME parameters unrecognizable
by client.
This commit makes a more selective encoding.

The error triggered a failure when sending attachments
with no-latin file names.

### CODE REVIEW

**Comment by javierag** [comment]
Unit test for this: https://github.com/zentyal/sogo/pull/174

**Comment by javierag** [comment]
The way to test it is to send attachments with no-latin filenames using Outlook. (Other clients use the filename extended header and are not affected by this)

File: sope-mime/NGMail/NGMimeMessageGenerator.m [hunk 1]
```diff
@@ -65,20 +65,68 @@ - (BOOL)isMultiValueCommaHeaderField:(NSString *)_headerField {
...
+    }
+
+  free(des);
+  return retval;
+}
+
+/*
+This function assumes the format of MIME header parameters.
+
+The no-MIME headers will continue to be correctly encoded, just it will be several q-encoded words separated as if they were header MIME parameters.
```
Comment (blaxter) [inline sope-mime/NGMail/NGMimeMessageGenerator.m]: There is no reason to exceed 80 chars length on comments.

File: sope-mime/NGMail/NGMimeMessageGenerator.m [hunk 2]
```diff
@@ -65,20 +65,68 @@ - (BOOL)isMultiValueCommaHeaderField:(NSString *)_headerField {
...
 }
 
-- (id)_escapeHeaderFieldValue:(NSData *)_data {
-  const char   *bytes  = NULL;
-  unsigned int length  = 0;
-  unsigned int desLen  = 0;
-  char         *des    = NULL;
-  unsigned int cnt;
-  BOOL         doEnc;
+int _encode_word(const char *bytes, unsigned int length, NSMutableData *destination)
```
Comment (blaxter) [inline sope-mime/NGMail/NGMimeMessageGenerator.m]: all sope (and sogo) code uses camel case style for method naming

**Comment by blaxter** [comment]
I've tested and it works perfect, awesome.

But I think the style of `{}` should be unify and changed (you've used both). All sope code uses (mostly, not always) the common:

    if (foo) {
        whatever
    } 

instead of sogo one:

    if (foo)
      {
        whatever
      }

**Comment by javierag** [comment]
I have fixed the style flaws found by Blaxter

**Comment by blaxter** [comment]
:+1:

**Comment by sixstone-qq** [comment]
:+1:

**Pull request merged**